Lib/fontTools/misc/configTools.py

```diff
@@ -106,45 +109,71 @@ class Options(Mapping):
     """
 
     __options: Dict[str, Option]
+    __cache: Set[Option]
 
-    def __init__(self) -> None:
+    def __init__(self, other: "Options" = None) -> None:
         self.__options = {}
+        self.__cache = set()
+        if other is not None:
+            for option in other.values():
+                self.register_option(option)
 
     def register(
         self,
         name: str,
         help: str,
         default: Any,
         parse: Callable[[str], Any],
-        validate: Callable[[Any], bool],
+        validate: Optional[Callable[[Any], bool]] = None,
     ) -> Option:
-        """Register a new option."""
-        return self.register_option(name, Option(help, default, parse, validate))
+        """Create and register a new option."""
+        return self.register_option(Option(name, help, default, parse, validate))
 
-    def register_option(self, name: str, option: Option) -> Option:
+    def register_option(self, option: Option) -> Option:
         """Register a new option."""
+        name = option.name
         if name in self.__options:
             raise ConfigAlreadyRegisteredError(name)
+        # sanity check option values are unique
+        assert option not in self.__cache
         self.__options[name] = option
+        self.__cache.add(option)
```

Comment:
Why do you need a cache, since you already have a dict() keyed by option name? Do you also want to ensure that the same option name is not represented by 2 different Option objects?

Reply:
exactly, right

Reply:
I want Option objects to be unique. There should only be one source of truth, the global options mapping. I made them frozen so one does't try to re register the same object twice under different names (can't change the name). I could even make the Option dataclass `eq=False` so it compares and hashes by id only (like `object`). I actually did that in an earlier iteration of this code, then undid that to not overcomplicate things.

Reply:
the cache is not actually necessary for I could check `option in self.values()`, but I think a set should speed it up

Reply:
The check `self.__options[option.name] is option` is both fast and more accurately locking down what should happen with options. What do you think?

Reply:
`self.__options.get(option.name) is option` -- i'll do that